src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -76,10 +76,10 @@ public void Equals_DifferentCalendars()
         {
             LocalDate start1 = new LocalDate(2000, 1, 1);
             LocalDate end1 = new LocalDate(2001, 6, 19);
-            // This is a really, really similar calendar to ISO - the dates could differ by week of year,
-            // but that's all.
-            LocalDate start2 = start1.WithCalendar(CalendarSystem.GetGregorianCalendar(1));
-            LocalDate end2 = end1.WithCalendar(CalendarSystem.GetGregorianCalendar(1));
+            // This is a really, really similar calendar to ISO, but we do distinguish.
+            // TODO: Should we?
```

Comment:
Absolutely. Options are:

- Have `CalendarSystem.Iso` and `CalendarySystem.Gregorian`, but return the same thing
- Have `CalendarSystem.Iso` and `CalendarySystem.Gregorian`, but return different calendars which are deemed unequal
- Have just `CalendarSystem.Iso`
- Have just `CalendarSystem.Gregorian`